scripts/tests/sanitycheck/test_testsuite_class.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python3
+# Copyright (c) 2020 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+# pylint: disable=line-too-long
+# pylint: disable=C0321
+'''
+This test file contains testcases for Testsuite class of sanitycheck
+'''
+import sys
+import os
+
+ZEPHYR_BASE = os.getenv("ZEPHYR_BASE")
+sys.path.insert(0, os.path.join(ZEPHYR_BASE, "scripts/sanity_chk"))
+
+from sanitylib import TestCase
+
+def test_testsuite_add_testcases(class_testsuite):
+    """ Testing add_testcase function of Testsuite class in sanitycheck """
+    # Test 1: Check the list of testcases after calling add testcases function is as expected
+    class_testsuite.SAMPLE_FILENAME = 'test_sample_app.yaml'
+    class_testsuite.TESTCASE_FILENAME = 'test_data.yaml'
+    class_testsuite.add_testcases()
+    expected_testcases = ['scripts/tests/sanitycheck/test_data/testcases/tests/test_b/kernel.fifo.timeout',
```

Comment:
You could test against something else, for example:

```
    expected_testcases = [
        'kernel.fifo',
        'kernel.fifo.poll',
        'kernel.fifo.usage',
        'kernel.fifo.usage.poll',
        'kernel.fifo.timeout',
        'kernel.fifo.timeout.poll',
        'sample.app_dev.external_lib'
    ]
    got = [t.id for t in class_testsuite.testcases.values()]
    assert sorted(got) == sorted(expected_testcases)
```

or you could define the common name of the path once and just use the variable everywhere...

It is not fun to review and see those paths, difficult to read and not really needed.

Reply:
I have cleaned this thing to just compare the last part of the testcases, instead of prefixing a string. Add_testcases function forms a dictionary of testcases where the key is testcase name coming from get_unique function (Which is the relative path to the testcase). But for making the paths look nicer, i am now just comparing the last part of the testcase name.